sklearn/ensemble/forest.py

```diff
@@ -354,6 +384,17 @@ def _validate_X_predict(self, X):
 
         return self.estimators_[0]._validate_X_predict(X, check_input=True)
 
+    def _validate_missing_mask(self, X, missing_mask=None):
```

Comment:
(At the moment, you dont really save computations, since you still need to performs boolean comparisons in this array, instead of doing float comparisons directly in X)